src/commands/rover/WhoisCommand.js

```diff
@@ -89,29 +89,30 @@ class WhoisCommand extends Command {
             json: true,
             simple: false
           })
-          pastNamesData.data.forEach(oldname => pastNames += `, ${oldname.name}`)
+          pastNamesData.data.forEach(oldname => { pastNames += `, ${oldname.name}` })
           if (pastNames) pastNames = pastNames.replace(', ', '')
         } catch (e) {}
-
-        let bc = 'Unknown'
+        const { cookie } = require('../../data/client.json')
+        let bc
         try {
           bc = await Cache.get(`bindings.${data.robloxId}`, 'bc')
-          if (!bc) {
+          if (!bc && cookie) {
             const response = await request({
-              uri: `https://groups.roblox.com/v1/users/${encodeURIComponent(data.robloxId)}/group-membership-status`,
+              uri: `https://premiumfeatures.roblox.com/v1/users/${data.robloxId}/validate-membership`,
               simple: false,
-              resolveWithFullResponse: true
+              json: true,
+              resolveWithFullResponse: true,
+              headers: {
+                cookie: `.ROBLOSECURITY=${cookie}`
+              }
             })
-
-            const membershipType = JSON.parse(response.body).membershipType
             bc = 'Regular'
-
-            if (membershipType === 4) {
+            if (response.body && response.statusCode === 200) {
               bc = 'Premium'
             }
 
             Cache.set(`bindings.${data.robloxId}`, 'bc', bc)
-          }
+          } else bc = 'Unknown'
```

Comment:
Could we perhaps have this hide if there is no cookie set? Just a way to reduce the "unknown" information on the whois embed.

Reply:
Done